### PR TITLE
standardise storage cost per byte

### PR DESCRIPTION
**Pull request by frisitano**
This PR standardises the cost of storage (per byte) across the different pallets that charge for storage usage in the reef runtime.  This includes transactions, evm and multisig.  The cost per byte has been standardised at 10 mREEF.

closes: #62

### CODE REVIEW

**Comment by Netherdrake** [comment]
For EVM, some backwards compatibility testing is required (ie. will the contract terminations refund storage deposit correctly).

**Review by Netherdrake — APPROVED** [review]

**Comment by frisitano** [comment]
I am currently in the process of exploring contract terminations.  Contract termination via `evm.selfdestruct` extrinsic seems reasonable.  My main concern is when a `selfdestruct` OPCODE is sent via an evm call as in this instance the destruction is handled by the EVM handler rather then the extrinsic logic.  Will report back with findings.

**Comment by frisitano** [comment]
TODO:

- Add tests for self destruct event emission via evm.call self destruct opcode
- refactor selfdestuct method such that `evm.call` (SUICIDE OPCODE) + `evm.selfdestruct` use the same contract destruction logic (https://github.com/reef-defi/reef-chain/blob/master/modules/evm/src/lib.rs#L890-L924).  Currently evm.call (SUICIDE) does not update `CodeInfos` and hence it could result in stale data being left behind which contributes to state bloat.

File: modules/evm/src/lib.rs [hunk 1]
```diff
@@ -678,11 +678,18 @@ pub mod module {
 
 impl<T: Config> Pallet<T> {
 	/// Process queued events
-	pub fn process_queued_events() {
+	pub fn process_queued_events() -> DispatchResult {
 		for event in Self::queued_events() {
+
+			if let Event::<T>::ContractSelfdestructed(contract, caller) = event {
```
Comment (frisitano) [inline modules/evm/src/lib.rs:684]: If contract removal fails the QueuedEvents will never be killed leaving events hanging.

Comment (frisitano) [reply modules/evm/src/lib.rs:684]: utilise `.drain(..)` to ensure vec is purged after every extrinsic - https://docs.substrate.io/rustdocs/latest/sp_std/vec/struct.Vec.html#method.drain

TODO: Add tests for failure case to ensure vec is cleared.

**Comment by frisitano** [comment]
TODO:

- Investigate failure to estimate evm costs in certain situations
- End to end testing

**Comment by frisitano** [comment]
This was not ready for merge - still had bugs relating to event queue not being purged properly.

**Pull request merged**